### PR TITLE
refactor: update example

### PR DESCRIPTION
**Pull request by theUnkownName**
- [x] Reduce to 2-3 indexes
- [x] Update to 0.9 
- [x] Switch to Jina types 
- [x] Update readme
- [ ] Update copyrights

### CODE REVIEW

File: fashion-example-query/app.py [hunk 1]
```diff
@@ -116,7 +103,6 @@ def download_data(target: dict, download_proxy=None):
 def index_generator(num_doc: int, target: dict):
     for j in range(num_doc):
         d = Document(content=target['index']['data'][j])
```
Comment (JoanFM) [inline fashion-example-query/app.py:105]: try to use `document` as context manager

File: fashion-example-query/app.py [hunk 2]
```diff
@@ -116,7 +103,6 @@ def download_data(target: dict, download_proxy=None):
...
-        d.update_id()
         label_int = target['index-labels']['data'][j][0]
```
Comment (JoanFM) [inline fashion-example-query/app.py:106]: maybe u should avoid indexing images of a label not belonging to the labels?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: fashion-example-query/README.md [hunk 3]
```diff
@@ -43,31 +43,19 @@ python app.py index
 First up we need to build up an index and then we'll search through this index when we use the query Flow later.
 So what is happening here?
 
-There are 10 categories in the fashion-mnist-data
+There are 10 categories in the fashion-mnist-data, but to simplify this example we will only use 3 categories:
 
     0	        T-shirt/top
     1	        Trouser
     2	        Pullover
```
Comment (cristianmtr) [inline fashion-example-query/README.md:50]: Any reason why we're sticking to `Pullover`, and not `pullover`?

File: fashion-example-query/README.md [hunk 4]
```diff
@@ -84,21 +72,21 @@ python app.py query
...
+So if we run this as it is, we will see our results filtered by Pullovers like this:
 
 ![alt text](.github/images/results.png "Results")
 
 
 ## QueryLanguage
 
 So the real magic here, if you compare it with the original HelloWorld, is that we are using QueryLanguage.
-This is happening in the index and query flow of each category, for example, for the ```indexer-dress.yml``` we have
+This is happening in the index and query flow of each category, for example, for the ```indexer-Pullover.yml``` we have
```
Comment (cristianmtr) [inline fashion-example-query/README.md:83]: It's kinda odd in filenames, no?

File: fashion-example-query/flows/index.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    shards: $JINA_SHARDS
+    separated_workspace: true
+    needs: [encode]
+  - name: indexer_trouser
+    uses: pods/indexer-trouser.yml
+    shards: $JINA_SHARDS
+    separated_workspace: true
+    needs: [encode]
+  - name: indexer_pullover
+    uses: pods/indexer-pullover.yml
```
Comment (cristianmtr) [inline fashion-example-query/flows/index.yml:22]: Ah, I see it is lowercase after all. So maybe make it consistent in the README

File: fashion-example-query/requirements.txt [hunk 6]
```diff
@@ -1,2 +1,6 @@
-jina[devel,torch,tensorflow,transformers]==0.7.12
 click
```
Comment (cristianmtr) [inline fashion-example-query/requirements.txt:1]: Specific version

**Review by cristianmtr — CHANGES_REQUESTED** [review]
some changes

File: fashion-example-query/requirements.txt (hunk 6 above)
Comment (theUnkownName) [reply fashion-example-query/requirements.txt:1]: I think I already changed this on the latest commit

Comment (cristianmtr) [reply fashion-example-query/requirements.txt:1]: I don't see it for `click`.

File: fashion-example-query/app.py [hunk 7]
```diff
@@ -61,7 +49,7 @@ def print_result(resp):
         result_html.append(f'<tr><td><img src="{vi}"/></td><td>')
         for kk in d.matches:
```
Comment (Yongxuanzhang) [inline fashion-example-query/app.py:50]: I think the variables within print_result() is not very readable. e.g. Shall we change d, vi, kk to more readable names?

**Review by Yongxuanzhang — COMMENTED** [review]
I think the variables within print_result() is not very readable. e.g. Shall we change d, vi, kk to more readable names?

File: fashion-example-query/app.py (hunk 7 above)
Comment (JoanFM) [reply fashion-example-query/app.py:50]: if u have a way to do so, u can also do that in the `hello-world` in the core for sure

Comment (Yongxuanzhang) [reply fashion-example-query/app.py:50]: I'm not sure what d, vi, kk, kmi represent. I would prefer some more descriptive ones, but not sure how to rename them.
And I also feel like 'd' is not a good naming convention. For example, if we use pdb to debug here, 'd' won't work. (https://stackoverflow.com/questions/15160946/python-debugger-tells-me-value-of-numpy-array-is-newest-frame).
What do you think?

Comment (JoanFM) [reply fashion-example-query/app.py:50]: vi is `uri` and kk is `match`

Comment (Yongxuanzhang) [reply fashion-example-query/app.py:50]: Yeah. I mean what are the full names of these abbreviations. 😄  
My thought is 
- d->data or query_data
- vi->data_uri or query_data_uri
- kk->match
- kmi->match_uri

Comment (JoanFM) [reply fashion-example-query/app.py:50]: Good lets go

Comment (theUnkownName) [reply fashion-example-query/app.py:50]: Sounds good, so we change it in this example and in the hello-workd example in core right?

File: fashion-example-query/flows/index.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+!Flow
+version: '1'
+with:
```
Comment (JoanFM) [inline fashion-example-query/flows/index.yml:3]: remove logserver option as it has been deprecated

**Review by JoanFM — COMMENTED** [review]
can u try to update to 0.9.20 (latest release) and removing the separated workspace parameter?

File: fashion-example-query/flows/query.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+pods:
+  - name: encode
+    uses: pods/encode.yml
+    parallel: $JINA_PARALLEL
+  - name: indexer_tshirt
+    uses: pods/indexer-tshirt.yml
+    shards: $JINA_SHARDS
+    needs: [encode]
+    polling: all
+    uses_after: pods/reduce.yml
```
Comment (JoanFM) [inline fashion-example-query/flows/query.yml:15]: make sure `reduce.yml` do proper sorting of results

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: fashion-example-query/flows/query.yml (hunk 9 above)
Comment (theUnkownName) [reply fashion-example-query/flows/query.yml:15]: But this is the same as before no? It was/is already giving good results

File: fashion-example-query/flows/index.yml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  compress_hwm: 1024
+pods:
+  - name: encode
+    uses: pods/encode.yml
+    parallel: $JINA_PARALLEL
```
Comment (cristianmtr) [inline fashion-example-query/flows/index.yml:8]: make sure you consistently use `shards` (`shards: $JINA_SHARDS_ENCOER` etc.)

**Review by cristianmtr — CHANGES_REQUESTED** [review]

File: fashion-example-query/flows/index.yml (hunk 10 above)
Comment (theUnkownName) [reply fashion-example-query/flows/index.yml:8]: By this you mean that in general we will use `parallel` as `JINA_SHARDS_ENCOER`?

File: fashion-example-query/flows/query.yml (hunk 9 above)
Comment (JoanFM) [reply fashion-example-query/flows/query.yml:15]: Yes I checked and is good

File: fashion-example-query/flows/index.yml (hunk 10 above)
Comment (cristianmtr) [reply fashion-example-query/flows/index.yml]: I meant:

1. Let's try to always use `shards` (and `parallel` will be reserved for other architecture options - parallel pods etc.)
2. Name the env. vars. with `JINA_SHARDS_COMPONENT` where `COMPONENT` is the specific component for which you're setting that number of shards.

**Review by JoanFM — APPROVED** [review]

File: fashion-example-query/README.md [hunk 11]
```diff
@@ -43,31 +43,19 @@ python app.py index
...
-
-We index the data updating updating the label
+We will create a separate index per category, so each category will have its own yml file and then we index the data updating updating the label
 
 ```
 d.tags.update({'label': get_mapped_label(label_int)})
 ```
 This is where you could tweak the code if you would like to see only one category.
-For example if you would like to see only dresses, you could do something like this
+For example if you would like to see only Pullovers, you could do something like this
```
Comment (cristianmtr) [inline fashion-example-query/README.md:58]:
```suggestion
For example if you would like to see only pullovers, you could do something like this
```

File: fashion-example-query/README.md [hunk 12]
```diff
@@ -84,21 +72,21 @@ python app.py query
 
 ## Results
 
-This is an example of the results after the query using all categories
+So if we run this as it is, we will see our results filtered by Pullovers like this:
```
Comment (cristianmtr) [inline fashion-example-query/README.md:75]:
```suggestion
So if we run this as it is, we will see our results filtered by pullovers like this:
```

File: fashion-example-query/README.md (hunk 3 above)
Comment (cristianmtr) [reply fashion-example-query/README.md:50]: @theUnkownName

File: fashion-example-query/README.md (hunk 12 above)
Comment (theUnkownName) [reply fashion-example-query/README.md]: Done

**Review by cristianmtr — APPROVED** [review]

**Pull request merged**